flake.nix

```diff
@@ -0,0 +1,68 @@
+{
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/22.11";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  outputs = { self, nixpkgs, flake-utils, }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        haskellPackages = pkgs.haskell.packages.ghc8107;
+      in {
+        formatter = pkgs.nixfmt;
+
+        ## `cabal test` is not run here because it requires internet
+        ## to fetch the dependencies
+        checks.default = pkgs.stdenv.mkDerivation {
+          name = "ormolu/hpack checks";
+          src = ./.;
+          buildInputs = with pkgs; [ findutils hpack ormolu ];
+          buildPhase = ''
+            hpack cooked-validators/package.yaml | grep -q 'is up-to-date'
+            hpack examples/package.yaml | grep -q 'is up-to-date'
+            hpack pirouette-plutusir | grep -q 'is up-to-date'
```

Comment:
Are Nix's `buildPhase`s running in `-e` mode?

Reply:
Then what is the point of those three lines?

Reply:
So if it never fails and `grep -q` doesn't print anything and `buildPhase` doesn't fail when `grep` fails, what is the point?

Reply:
I guess that or explicit failure, whichever works best!

Reply:
That is precisely the one that's not specified in POSIX: it can be ran in the current Shell environment or in a sub-Shell up to implementation. In bash I think it's in the same Shell, so we should be fine.

Reply:
That does sound more predictable, yes!